src/frontend/src/pages/GanttPage/GanttChart.tsx

```diff
@@ -92,8 +90,7 @@ const GanttChart = ({
                 end={endDate}
                 isEditMode={isEditMode}
                 onExpanderClick={(newTask) => {
-                  const newTasks = ganttTasks.map((task) => (newTask.id === task.id ? { ...newTask, teamName } : task));
-                  setGanttTasks(newTasks);
+                  ganttTasks = ganttTasks.map((task) => (newTask.id === task.id ? { ...newTask, teamName } : task));
```

Comment:
this works but it's a bit fucky because you're passing in a mutable variable as a prop. It'd be good if we could make this method you've written here a callback that's passed in as a prop to the GanttChart component. I think if we do that, we shouldn't need ganttTasks as a prop to this method, so you can remove that and it should solve the problem of that prop being mutable as well